ibm/service/directlink/resource_ibm_dl_gateway.go

```diff
@@ -1070,7 +1100,8 @@ func resourceIBMdlGatewayRead(d *schema.ResourceData, meta interface{}) error {
 	}
 	log.Printf("[INFO] Calling getgateway api: %s", dtype)
 
-	instance, response, err := directLink.GetGateway(getOptions)
+	instanceIntf, response, err := directLink.GetGateway(getOptions)
+	instance := instanceIntf.(*directlinkv1.GetGatewayResponse)
```

Comment:
Can we add some nil check before instanceIntf.(*directlinkv1.GetGatewayResponse) to suppress any nil pinter execeptions

Reply:
Done